pkg/services/dashboards/database/database.go

```diff
@@ -962,17 +965,236 @@ func (d *dashboardStore) FindDashboards(ctx context.Context, query *dashboards.F
 
 	sql, params := sb.ToSQL(limit, page)
 
+	// Only use modified search for non-empty search queries, otherwise it's just listing and new query can't help there yet.
```

Comment:
It would be great if we could put as much of this into a separate function as possible. One of the goals of panel title search was to reduce the number of search offerings and therefore simplify our codebase. We will not be able to do that (yet) by introducing alt search. However, isolating it as much as possible should help.